compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2814,6 +2814,7 @@ private VariableDefinitionNode createBLangVarDef(Location location,
             case LIST_BINDING_PATTERN:
                 initializeBLangVariable(variable, typedBindingPattern.typeDescriptor(), initializer,
                         qualifierList);
+                variable.pos = location;
```

Comment:
Is this fix correct? Don't we need to have a fix like below?
variable.pos = getPosition(bindingPattern);

Reply:
Check this again.

Reply:
@mohanvive 

Considering the example `var [fName, fBody] = parseField(l);`, 

- `getPosition(bindingPattern);` would provide the position of `var <start>[fName, fBody]<end> = parseField(l);`

- `getPosition(typedBindingPattern)` would provide the position of `<start>var [fName, fBody]<end> = parseField(l);`

- `location` would provide the position for the entire var declaration node `<start>var [fName, fBody] = parseField(l)<end>`

This approach is also being used for the `WILDCARD_BINDING_PATTERN` case at https://github.com/ballerina-platform/ballerina-lang/blob/8adfc0ac5b12d60c70d33893a15e49e7eba2920e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java#L2798

Reply:
Used a different approach to set the position of `variable` as `var <start>[fName, fBody] = parseField(l);<end>`